src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -782,6 +799,38 @@ namespace AirLoopHVACDOAS {
 
                 thisDOAS.m_AirLoopDOASNum = AirLoopDOASNum - 1;
                 state.dataAirLoopHVACDOAS->airloopDOAS.push_back(thisDOAS);
+
+                // ensure the inlet node should be an outdoor air node
+                bool OAnodeFound = false;
+                for (int OutsideAirNodeNum = 1; OutsideAirNodeNum <= state.dataOutAirNodeMgr->NumOutsideAirNodes; ++OutsideAirNodeNum) {
+                    if (thisDOAS.m_InletNodeNum == state.dataOutAirNodeMgr->OutsideAirNodeList(OutsideAirNodeNum)) {
+                        OAnodeFound = true;
+                        break;
+                    }
+                }
+                if (!OAnodeFound) {
+                    ShowSevereError(state,
+                                    format("Inlet node ({}) is not one of OutdoorAir:Node in {} = {}",
+                                           state.dataLoopNodes->NodeID(thisDOAS.m_InletNodeNum),
+                                           CurrentModuleObject,
+                                           thisDOAS.Name));
+                    errorsFound = true;
+                }
+                // Ensure the outlet node is the splitter inlet node
+                bool OutletnodeFound = false;
+                if (thisDOAS.m_OutletNodeNum == thisDOAS.m_CompPointerAirLoopSplitter->InletNodeNum) {
+                    OutletnodeFound = true;
+                }
+                if (!OutletnodeFound) {
```

Comment:
Minor, but `OutletnodeFound` isn't necessary here, just collapse this into a single `if`.

Reply:
@mjwitte Done.